common/djangoapps/student/views.py

```diff
@@ -1677,8 +1677,13 @@ def create_account_with_params(request, params):
     # when the account is created via the browser and redirect URLs.
     should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
 
-    if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
+    # if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
+    #     params["password"] = pipeline.make_random_password()
+
+    if params.get('third_party') and pipeline.running(request):
```

Comment:
@awaisdar001  we can remove `pipeline.running(request)` check if we are sure that pipe line is getting checked only for identification of whether this request is made using third party auth.